numismatic/feeds/base.py

```diff
@@ -103,23 +103,32 @@ def subscribe(self, assets, currencies, channels, exchange=None,
         currencies = self._validate_parameter('currencies', currencies)
         channels = self._validate_parameter('channels', channels)
         subscriptions = {}
-        for symbol, channel in product(self._get_pairs(assets, currencies),
-                                       channels):
-            if self._websocket_client_class is not None:
-                if self.websocket_client is None:
-                    self.websocket_client = self._websocket_client_class()
-                subscription = self.websocket_client.subscribe(symbol, channel)
-            elif self._rest_client_class is not None:
-                channel_method = getattr(self, f'get_{channel.lower()}')
-                rest_client = self._rest_client_class()
-                subscription = rest_client.subscribe(symbol, channel_method,
-                                                     interval=interval,
-                                                     exchange=exchange)
-            else:
-                raise ValueError('No subscribe() method found.')
-            subscriptions[subscription.market_name] = subscription
+        for asset, currency, channel in product(assets, currencies, channels):
+            subscription = \
+                self._subscribe(asset, currency, channel, exchange=exchange,
+                                interval=interval)
+            subscriptions[subscription.topic] = subscription
         return subscriptions
 
+    def _subscribe(self, asset, currency, channel, exchange=None, 
+                   interval=1.0):
+        if self._websocket_client_class is not None:
```

Comment:
In the implementation below, if WS client is defined, it will ignore the Rest client class, implying that if you have both WS client and Rest Client defined, and (for argument sake) the subscribe you are interested in is defined on the Rest client, it will get ignored here?

Reply:
I agree with you. We would need to define a mechanism to resolve that. I'll create an issue for that. This has existed in the codebase for a while and doesn't need to be fixed in this PR.